### PR TITLE
Add check that there is no PackedDataset while building ConcatDataset

### PR DESCRIPTION
**Pull request by krammnic**
#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [X] fix a bug
- [X] update tests and/or documentation
- [ ] other (please add here)

Please link to any issues this PR addresses.

#### Changelog
What are the changes made in this PR?
* Add check that there is no PackedDataset while building ConcatDataset #1708

#### Test plan
Please make sure to do each of the following if applicable to your PR. If you're unsure about any one of these just ask and we will happily help. We also have a [contributing page](https://github.com/pytorch/torchtune/blob/main/CONTRIBUTING.md) for some guidance on contributing.

- [X] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [X] add [unit tests](https://github.com/pytorch/torchtune/tree/main/tests/torchtune) for any new functionality
- [X] update [docstrings](https://github.com/pytorch/torchtune/tree/main/docs/source) for any new or updated methods or classes
- [X] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [X] manually run any new or modified recipes with sufficient proof of correctness
- [ ] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

#### UX
If your function changed a public API, please add a dummy example of what the user experience will look like when calling it.
Here is a [docstring example](https://github.com/pytorch/torchtune/blob/6a7951f1cdd0b56a9746ef5935106989415f50e3/torchtune/modules/vision_transformer.py#L285)
and a [tutorial example](https://pytorch.org/torchtune/main/tutorials/qat_finetune.html#applying-qat-to-llama3-models)

- [ ] I did not change any public API
- [X] I have added an example to docs or docstrings

### CODE REVIEW

**Comment by krammnic** [comment]
Actually pretty strange, made pre-commit run for all required files and no changes probably. But let's see CI/CD here

File: torchtune/datasets/_concat.py [hunk 1]
```diff
@@ -64,6 +68,13 @@ class ConcatDataset(Dataset):
 
     def __init__(self, datasets: List[Dataset]):
         self._datasets: List[Dataset] = datasets
+
+        for dataset in self._datasets:
+            if isinstance(dataset, PackedDataset):
+                raise ValueError(
+                    "ConcatDataset can't proceed instances of PackedDataset."
```
Comment (ebsmothers) [inline torchtune/datasets/_concat.py:75]: nit
```suggestion
                    "ConcatDataset can't process instances of PackedDataset."
```

**Review by ebsmothers — COMMENTED** [review]
Thanks for fixing this! One small nit, but otherwise this looks good. Once CI is green I'll stamp and merge

**Comment by krammnic** [comment]
@ebsmothers Fixed lint. Now should be fine.

**Comment by krammnic** [comment]
Yeah, lint passed.

**Comment by krammnic** [comment]
@ebsmothers can be merged now.

**Review by ebsmothers — APPROVED** [review]

**Pull request merged**